src/basic/mount-util.c

```diff
@@ -217,6 +217,18 @@ int path_is_mount_point(const char *t, const char *root, int flags) {
         if (path_equal(t, "/"))
                 return 1;
 
+        /* basename() does not work if the path has a trailing slash.
+         * Make sure it's not there. */
+        if (endswith(t, "/")) {
+                unsigned len;
+
+                len = strlen(t);
+                while (len > 0 && t[len-1] == '/')
+                        len--;
```

Comment:
we should probably add some helper for dropping trailing characters to string-util.h I figure and then use this here. And I figure we could probably replace a few other pieces of code like this too...

Reply:
OK.